### PR TITLE
docs: Add natspec docs to contracts

### PR DESCRIPTION
**Pull request by miguelmtzinf**

### CODE REVIEW

File: contracts/bridges/BridgeExecutorBase.sol [hunk 1]
```diff
@@ -1,47 +1,61 @@
...
- *
- * @notice This is an abstract contract that implements basic governance executor functionality.
- * It does not implement an external queue function, this should instead be done in the inheriting
- * contract with proper access control.
+ * @notice Abstract contract that implements basic executor functionality
+ * @dev It does not implement an external `queue` function. This should instead be done in the inheriting
+ * contract with proper access control
  */
 abstract contract BridgeExecutorBase is IExecutorBase {
+  // Minimum allowed grace period, which reduces the risk of having actions set expired due to network congestion
```
Comment (Zer0dot) [inline contracts/bridges/BridgeExecutorBase.sol:14]:
```suggestion
// Minimum allowed grace period, which reduces the risk of having an action set expire due to network congestion
```
Just a small grammar fix.

File: contracts/bridges/ArbitrumBridgeExecutor.sol [hunk 2]
```diff
@@ -4,13 +4,30 @@ pragma solidity 0.8.10;
 import {AddressAliasHelper} from '../dependencies/arbitrum/AddressAliasHelper.sol';
 import {L2BridgeExecutor} from './L2BridgeExecutor.sol';
 
+/**
+ * @title ArbitrumBridgeExecutor
+ * @author Aave
+ * @notice Implementation of the Arbitrum Bridge Executor, able to receive cross-chain transactions from Ethereum
+ * @dev Queuing an ActionSet into this Executor can only be done by the L2 Address Alias of the L1 EthereumGovernanceExecutor
```
Comment (Zer0dot) [inline contracts/bridges/ArbitrumBridgeExecutor.sol:11]:
```suggestion
 * @dev Queuing an ActionsSet into this Executor can only be done by the L2 Address Alias of the L1
```

File: contracts/bridges/OptimismBridgeExecutor.sol [hunk 3]
```diff
@@ -4,9 +4,18 @@ pragma solidity 0.8.10;
 import {ICrossDomainMessenger} from '../interfaces/ICrossDomainMessenger.sol';
 import {L2BridgeExecutor} from './L2BridgeExecutor.sol';
 
+/**
+ * @title OptimismBridgeExecutor
+ * @author Aave
+ * @notice Implementation of the Optimism Bridge Executor, able to receive cross-chain transactions from Ethereum
+ * @dev Queuing an ActionSet into this Executor can only be done by the Optimism L2 Cross Domain Messenger and having
```
Comment (Zer0dot) [inline contracts/bridges/OptimismBridgeExecutor.sol:11]:
```suggestion
 * @dev Queuing an ActionSet into this Executor can only be done by the Optimism L2 Cross Domain Messenger and having
```

File: contracts/bridges/PolygonBridgeExecutor.sol [hunk 4]
```diff
@@ -4,21 +4,55 @@ pragma solidity 0.8.10;
 import {IFxMessageProcessor} from '../interfaces/IFxMessageProcessor.sol';
 import {BridgeExecutorBase} from './BridgeExecutorBase.sol';
 
+/**
+ * @title PolygonBridgeExecutor
+ * @author Aave
+ * @notice Implementation of the Polygon Bridge Executor, able to receive cross-chain transactions from Ethereum
+ * @dev Queuing an ActionSet into this Executor can only be done by the FxChild and having the EthereumGovernanceExecutor
```
Comment (Zer0dot) [inline contracts/bridges/PolygonBridgeExecutor.sol:11]:
```suggestion
 * @dev Queuing an ActionsSet into this Executor can only be done by the FxChild and having the EthereumGovernanceExecutor
 ```

File: contracts/bridges/BridgeExecutorBase.sol [hunk 5]
```diff
@@ -251,12 +266,13 @@ abstract contract BridgeExecutorBase is IExecutorBase {
...
-   * @param targets list of contracts called by each action's associated transaction
-   * @param values list of value in wei for each action's  associated transaction
-   * @param signatures list of function signatures (can be empty) to be used when created the callData
-   * @param calldatas list of calldatas: if associated signature empty, calldata ready, else calldata is arguments
-   * @param withDelegatecalls boolean, true = transaction delegatecalls the target, else calls the target
+   * @notice Queue an ActionsSet
+   * @dev If a signature is empty, calldata is used for the execution, calldata is appended to signature otherwise
+   * @param targets Array of targets to be called by the actions set
+   * @param values Array of values to pass in each call by the actions set
+   * @param signatures Array of function signatures to encode in each call by the actions (can be empty)
```
Comment (Zer0dot) [inline contracts/bridges/BridgeExecutorBase.sol:273]:
```suggestion
   * @param signatures Array of function signatures to encode in each call (can be empty)
```

File: contracts/bridges/BridgeExecutorBase.sol [hunk 6]
```diff
@@ -251,12 +266,13 @@ abstract contract BridgeExecutorBase is IExecutorBase {
...
+   * @param calldatas Array of calldata to pass in each call by the actions set
```
Comment (Zer0dot) [inline contracts/bridges/BridgeExecutorBase.sol:274]:
```suggestion
   * @param calldatas Array of calldata to pass in each call (can be empty)
```

File: contracts/bridges/BridgeExecutorBase.sol [hunk 7]
```diff
@@ -251,12 +266,13 @@ abstract contract BridgeExecutorBase is IExecutorBase {
...
+   * @param withDelegatecalls Array of whether to delegatecall for each call of the actions set
```
Comment (Zer0dot) [inline contracts/bridges/BridgeExecutorBase.sol:275]:
```suggestion
   * @param withDelegatecalls Array of whether to delegatecall for each call
```

File: contracts/bridges/L2BridgeExecutor.sol [hunk 8]
```diff
@@ -1,20 +1,35 @@
...
+import {IL2BridgeExecutor} from '../interfaces/IL2BridgeExecutor.sol';
 import {BridgeExecutorBase} from './BridgeExecutorBase.sol';
 
-abstract contract L2BridgeExecutor is BridgeExecutorBase {
+/**
+ * @title L2BridgeExecutor
+ * @author Aave
+ * @notice Abstract contract that implements bridge executor functionality for L2
+ * @dev It does not implement the `onlyEthereumGovernanceExecutor` modifier. This should instead be done in the inheriting
+ * contract with proper configuration and adjustments depending of the L2
```
Comment (Zer0dot) [inline contracts/bridges/L2BridgeExecutor.sol:12]:
```suggestion
 * contract with proper configuration and adjustments depending on the L2
```

File: contracts/bridges/L2BridgeExecutor.sol [hunk 9]
```diff
@@ -1,20 +1,35 @@
...
+ */
+abstract contract L2BridgeExecutor is BridgeExecutorBase, IL2BridgeExecutor {
+  // Address of the Ethereum Governance Executor, able to queue actions sets
```
Comment (Zer0dot) [inline contracts/bridges/L2BridgeExecutor.sol:15]:
```suggestion
  // Address of the Ethereum Governance Executor, which should be able to queue actions sets
```

File: contracts/bridges/L2BridgeExecutor.sol [hunk 10]
```diff
@@ -1,20 +1,35 @@
...
 
-  event EthereumGovernanceExecutorUpdate(
-    address previousEthereumGovernanceExecutor,
-    address newEthereumGovernanceExecutor
-  );
-
-  error UnauthorizedEthereumExecutor();
-
+  /**
+   * @dev Only the Ethereum Governance Executor can call functions marked by this modifier.
```
Comment (Zer0dot) [inline contracts/bridges/L2BridgeExecutor.sol:19]:
```suggestion
   * @dev Only the Ethereum Governance Executor should be able to call functions marked by this modifier.
```

This and the other comment change I requested are just specifying that this is the intended behavior, but it isn't actually enforced here.

File: contracts/interfaces/IExecutorBase.sol [hunk 11]
```diff
@@ -66,15 +59,15 @@ interface IExecutorBase {
...
-   * @param signatures list of function signatures (can be empty) to be used when created the callData
-   * @param calldatas list of calldatas: if associated signature empty, calldata ready, else calldata is arguments
-   * @param withDelegatecalls boolean, true = transaction delegatecalls the target, else calls the target
-   * @param executionTime the time these actions can be executed
+   * @param targets Array of targets to be called by the actions set
+   * @param values Array of values to pass in each call by the actions set
+   * @param signatures Array of function signatures to encode in each call by the actions set
+   * @param calldatas Array of calldata to pass in each call by the actions set
+   * @param withDelegatecalls Array of whether to delegatecall for each call of the actions set
+   * @param executionTime The timestamp in which this actions set can be executed
```
Comment (Zer0dot) [inline contracts/interfaces/IExecutorBase.sol:67]:
```suggestion
   * @param executionTime The timestamp at which this actions set can be executed
 ```

File: contracts/interfaces/IExecutorBase.sol [hunk 12]
```diff
@@ -84,171 +77,174 @@ interface IExecutorBase {
...
+   * @dev It does not affect to actions set that are already queued
+   * @param delay The value of the delay (in seconds)
    **/
   function updateDelay(uint256 delay) external;
 
   /**
-   * @dev Set the grace period, the time after a delay during which a proposal can be
-   * executed. This must be greater than the minimum of 10 minutes (600).
-   * @param gracePeriod The gracePeriod in seconds
+   * @notice Update the grace period, time after the execution time during which an actions set can be executed
```
Comment (Zer0dot) [inline contracts/interfaces/IExecutorBase.sol:158]:
```suggestion
   * @notice Update the grace period, the period after the execution time during which an actions set can be executed
```

**Review by Zer0dot — COMMENTED** [review]
Just a few changes, otherwise lgtm!

File: contracts/bridges/L2BridgeExecutor.sol (hunk 10 above)
Comment (miguelmtzinf) [reply contracts/bridges/L2BridgeExecutor.sol:19]: Yes, good point

**Pull request merged**